### ISSUE TITLE
CACHE_SIZE is set weirdly high for all deployments

### ISSUE THREAD

**Issue by datenhahn** [milvus-io/milvus-helm#355]
Hi,

the cache_size is set in the charts to:

https://github.com/milvus-io/milvus-helm/blob/master/charts/milvus/templates/querycoord-deployment.yaml#L64
```
        - name: CACHE_SIZE
          valueFrom:
            resourceFieldRef:
              divisor: 1Gi
              resource: limits.memory
```

In the default Config it says:

https://github.com/milvus-io/milvus/blob/master/configs/milvus.yaml#L164

"The `cacheSize` must be less than system memory size."

As the helm-chart sets the memory to the absolutely maximum of memory a pod can ever request (the limit) I am wondering if this can hurt milvus stability. Or is this fine and the comment in the default config needs to be adjusted instead?

**Comment by LoveEachDay** [comment]
@datenhahn It's recommended to set the pod memory limits explicitly. 
If you doesn't set the memory explicitly, it will default to the memory of the kubernetes worker pod which pod resides in.

**Comment by datenhahn** [comment]
I don't know about the default behavior if the limit is not set, I am only using pod with explicitly set limits.

If I set the memory limit with `resources.limits` on a pod to lets say 8GB, then the container has exactly 8GB memory as absolute maximum available. In the helmchart the cache_size is set to "8", which means, the cache will grab all available 8GB and not leave anything for operation system processes, etc.. Either this is fine, because milvus adjusts the value correctly (e.g. internally only takes 70% of the passed "CACHE_SIZE") or this is potentially an issue, as milvus might eat up all OS memory with the cache and then crash.

The comment in the default values yaml file indicates that the memory size of the cache should be less than the total memory available, so I guess milvus does NOT adjust the cache memory used by itself. The user must pass a reasonable value (I did not find any sources about what is a reasonable number, but I would assume something about 50-70% of the system memory?)

https://github.com/milvus-io/milvus/blob/master/configs/milvus.yaml#L164

"The cacheSize must be less than system memory size."
